hexrd/transforms/new_capi/xf_new_capi.py

```diff
@@ -28,142 +29,364 @@
+        Default is None
+    output_ref : bool, optional
+        If True, prepends the apparent bragg angle and azimuth with respect to
+        the SAMPLE FRAME (ignoring effect of non-zero tvec_c)
+
+    Returns
+    -------
+    array_like
+        (n, 2) ndarray containing the (tth, eta) pairs associated with each
+        (x, y) associated with gVecs
+    array_like
+        (n, 3) ndarray containing the associated G vector directions in the
+        LAB FRAME
+    array_like, optional
+        if output_ref is True
+    """
+    # TODO: in the C library beam vector and eta vector are expected. However
+    # we receive rmat_b. Please check this!
     #
-    # It also seems that the output_ref version is not present as the argument gets
-    # ignored
+    # It also seems that the output_ref version is not present as the argument
+    # gets ignored
 
     rmat_b = rmat_b if rmat_b is not None else cnst.identity_3x3
 
-    # the code seems to ignore this argument, assume output_ref == True not implemented
-    assert not output_ref
+    # the code seems to ignore this argument, assume output_ref == True not
+    # implemented
+    assert not output_ref, 'output_ref not implemented'
 
     if distortion is not None:
         xy_d = distortion.unwarp(xy_d)
 
-    xy_d  = np.ascontiguousarray( np.atleast_2d(xy_d) )
-    rmat_d = np.ascontiguousarray( rmat_d )
-    rmat_s = np.ascontiguousarray( rmat_s )
-    tvec_d  = np.ascontiguousarray( tvec_d.flatten() )
-    tvec_s  = np.ascontiguousarray( tvec_s.flatten() )
-    tvec_c  = np.ascontiguousarray( tvec_c.flatten() )
-    beam_vec = np.ascontiguousarray( (-rmat_b[:,2]).flatten() )
-    eta_vec  = np.ascontiguousarray( rmat_b[:,0].flatten() ) #check this!
-    return _impl.detectorXYToGvec(xy_det,
-                                  rmat_d, rmat_s,
-                                  tvec_d, tvec_s, tvec_c,
-                                  beam_vec, eta_vec)
-
-
-#@xf_api
-def oscillAnglesOfHKLs(hkls, chi, rMat_c, bMat, wavelength,
-                       vInv=None, beamVec=cnst.beam_vec, etaVec=cnst.eta_vec):
+    xy_d = np.ascontiguousarray(np.atleast_2d(xy_d))
+    rmat_d = np.ascontiguousarray(rmat_d)
+    rmat_s = np.ascontiguousarray(rmat_s)
+    tvec_d = np.ascontiguousarray(tvec_d.flatten())
+    tvec_s = np.ascontiguousarray(tvec_s.flatten())
+    tvec_c = np.ascontiguousarray(tvec_c.flatten())
+    beam_vec = np.ascontiguousarray((-rmat_b[:, 2]).flatten())
+    eta_vec = np.ascontiguousarray(rmat_b[:, 0].flatten())  # check this!
+    return _impl.detectorXYToGvec(
+        xy_d, rmat_d, rmat_s, tvec_d, tvec_s, tvec_c, beam_vec, eta_vec
+    )
+
+
+# @xf_api
+def oscillAnglesOfHKLs(
+    hkls,
+    chi,
+    rMat_c,
+    bMat,
+    wavelength,
+    vInv=None,
+    beamVec=cnst.beam_vec,
+    etaVec=cnst.eta_vec,
+):
     # this was oscillAnglesOfHKLs
     hkls = np.array(hkls, dtype=float, order='C')
     if vInv is None:
         vInv = np.ascontiguousarray(vInv_ref.flatten())
```

Comment:
I think this is also a bug, vInv_ref doesn't seem to be defined.  But I don't think this funciton is used anyway,